apis/supervisor/idp/v1alpha1/types_oidcidentityprovider.go.tmpl

```diff
@@ -138,6 +138,17 @@ type OIDCClaims struct {
 	// the ID token.
 	// +optional
 	Username string `json:"username"`
+
+	// AdditionalClaimMappings allows for additional arbitrary upstream claim values to be mapped into the
+	// "additionalClaims" claim of the ID tokens generated by the Supervisor. This should be specified as a map of
+	// new claim names as the keys, and upstream claim names as the values. These new claim names will be nested
+	// under the top-level "additionalClaims" claim in ID tokens generated by the Supervisor when this
+	// OIDCIdentityProvider was used for user authentication. This feature is not required for using the Supervisor to
+	// provide authentication for Kubernetes clusters, but can be used when using the Supervisor for other
+	// authentication purposes. When this map is empty, the "additionalClaims" claim will be excluded from the ID
+	// tokens generated by the Supervisor.
```

Comment:
This API doc should also explain how this claim does or does not get included in ID tokens for `OIDCClients`.

Reply:
We should discuss whether `additionalClaims` requires a new scope.